### PR TITLE
Auto-reload TLS certificates on SIGHUP

### PR DESCRIPTION
**Pull request by codysoyland**
## Overview

This patch allows you to update expiring TLS certificates in-place without restarting Pilosa. It listens for SIGHUP and reloads the certificate/key from disk when triggered.

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Add appropriate changelog label to PR (if applicable).

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Make sure PR is tagged with appropriate changelog label.

### CODE REVIEW

**Comment by jaffee** [comment]
You've probably considered this, so I'll be interested to hear your thoughts, but I think we should copy in this dependency rather than importing it.

I was waffling, but what sealed the deal was that I noticed it used a bare `log` statement which we wouldn't be able to control through our normal means.

**Comment by codysoyland** [comment]
I was waffling too, but I felt that fewer LOC were better and would avoid having to distribute the third party license, but you make a good point about the logger. I will update it.

File: server/tlsconfig.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 Pilosa Corp.
```
Comment (jaffee) [inline server/tlsconfig.go]: should this have the other copyright/license info in addition to or in place of ours?

Comment (codysoyland) [reply server/tlsconfig.go]: Yeah good call, better to be more explicit. Updated.

**Pull request merged**